javascript/index.js

```diff
@@ -23,18 +24,139 @@
+  document.querySelector("#mashedPotatoes").innerHTML += `<li>${step0}</li>`;
+
+  getInstruction('mashedPotatoes', 1, (step1) => {
+    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step1}</li>`;
+
+    getInstruction('mashedPotatoes', 2, (step2) => {
+      document.querySelector("#mashedPotatoes").innerHTML += `<li>${step2}</li>`;
+
+      getInstruction('mashedPotatoes', 3, (step3) => {
+        document.querySelector("#mashedPotatoes").innerHTML += `<li>${step3}</li>`;
+
+        getInstruction('mashedPotatoes', 4, (step4) => {
+          document.querySelector("#mashedPotatoes").innerHTML += `<li>${step4}</li>`;
+          document.querySelector("#mashedPotatoes").innerHTML += `<li>Mashed potatoes are ready!</li>`;
+          document.querySelector("#mashedPotatoesImg").style.display = "block"
+        });
+      });
+    });
+  });
+});
+
+
 
 // Iteration 2 - using promises
-// ...
+
+
+
+obtainInstruction('steak', 0)
+  .then((step0) => {
+    document.querySelector("#steak").innerHTML += `<li>${step0}</li>`;
+    return obtainInstruction('steak', 1);
+  })
+  .then((step1) => {
+    document.querySelector("#steak").innerHTML += `<li>${step1}</li>`;
+    return obtainInstruction('steak', 2);
+  })
+  .then((step2) => {
+    document.querySelector("#steak").innerHTML += `<li>${step2}</li>`;
+    return obtainInstruction('steak', 3);
+  })
+  .then((step3) => {
+    document.querySelector("#steak").innerHTML += `<li>${step3}</li>`;
+    return obtainInstruction('steak', 4);
+  })
+  .then((step4) => {
+    document.querySelector("#steak").innerHTML += `<li>${step4}</li>`;
+    return obtainInstruction('steak', 5);
+  })
+  .then((step5) => {
+    document.querySelector("#steak").innerHTML += `<li>${step5}</li>`;
+    return obtainInstruction('steak', 6);
+  })
+  .then((step6) => {
+    document.querySelector("#steak").innerHTML += `<li>${step6}</li>`;
+    return obtainInstruction('steak', 7);
+    
+  })
+  .then((step7) => {
+    document.querySelector("#steak").innerHTML += `<li>${step7}</li>`;
+    document.querySelector("#steak").innerHTML += "<li>Steak is ready!</li>";
+
+  })
+  .then(()=>document.querySelector("#steakImg").style.display = "block")
+  .catch((error) => {
+    console.error("Error obtaining instructions:", error);
+  });
+
 
 // Iteration 3 using async/await
-// ...
+
+
+
+async function makeBroccoli() {
+  const broccoliContainer = document.querySelector("#broccoli");
+  try {
+    for (let i = 0; i < broccoli.length; i++) {
+      const step = await obtainInstruction('broccoli', i);
+      broccoliContainer.innerHTML += `<li>${step}</li>`;
+    }
+
+    broccoliContainer.innerHTML += "<li>Broccoli is ready!</li>";
+    document.querySelector("#broccoliImg").style.display = "block"
+  } catch (error) {
+    console.error("Error obtaining instructions:", error);
+  }
+}
+
+makeBroccoli();
+
+
+// Bonus 1
+
+
 
 // Bonus 2 - Promise all
-// ...
\ No newline at end of file
+
+
```

Comment:
Aamir, your solution works fine, but here's another way that you could use next time:
```
// Bonus 2
// Using Promise.all() with async/await syntax:

const pr0 = obtainInstruction('broccoli', 0);
const pr1 = obtainInstruction('broccoli', 1);
const pr2 = obtainInstruction('broccoli', 2);
const pr3 = obtainInstruction('broccoli', 3);
const pr4 = obtainInstruction('broccoli', 4);
const pr5 = obtainInstruction('broccoli', 5);
const pr6 = obtainInstruction('broccoli', 6);

async function makeBrusselSprouts() {
  try {
    const values = await Promise.all([pr0, pr1, pr2, pr3, pr4, pr5, pr6]);

    values.forEach((instruction) => {
      document.querySelector('#brusselsSprouts').innerHTML += `<li>${instruction}</li>`;
    });

    document.querySelector('#brusselsSprouts').innerHTML += `<li>Brussels sprouts are ready!</li>`;
    document.querySelector('#brusselsSproutsImg').removeAttribute('hidden');
  } catch (error) {
    console.log(err);
  }
}

makeBrusselSprouts();
```